### PR TITLE
Add isBounded property so items could be constrained within grid

### LINKED ISSUE #546: How do I constrain elements to their container?
- [ ] Bug
- [ ] Feature Request
- [x] Question

Just using the default example code. I would expect `b` to not break out of the container (like in the demos).

grid-overlap

What am I doing wrong?

### PR DESCRIPTION
**Pull request by artembykov**
This PR fixes #546

### CODE REVIEW

File: lib/GridItem.jsx [hunk 1]
```diff
@@ -289,24 +309,41 @@ export default class GridItem extends React.Component {
...
-        case 'onDrag':
+        case 'onDrag': {
           if (!this.state.dragging) throw new Error('onDrag called before onDragStart.');
-          newPosition.left = this.state.dragging.left + deltaX;
-          newPosition.top = this.state.dragging.top + deltaY;
+
+          let top = this.state.dragging.top + deltaY;
+          let left = this.state.dragging.left + deltaX;
+
+          if (this.props.isBounded) {
```
Comment (artembykov) [inline lib/GridItem.jsx]: Constraints applied here

**Comment by artembykov** [comment]
@STRML, what do you think about that PR? To clarify things a little bit more, these changes add a property which controls wether placeholders should be bounded within grid while dragging items around

**Comment by mpsk** [comment]
@artembykov, there are few file conflicts.
@STRML please, pay attention to this PR and merge as a fix is really important.

**Comment by aslafy-z** [comment]
Any ETA on getting this merged ?

**Comment by artembykov** [comment]
ping @STRML
I've rebased on master today with one small perf improvement.

**Comment by aslafy-z** [comment]
https://github.com/STRML/react-grid-layout/pull/549

**Comment by abenhamdine** [comment]
@artembykov there are file conflicts again.

**Comment by artembykov** [comment]
> @artembykov there are file conflicts again.

Resolved!

**Comment by abenhamdine** [comment]
@STRML any chance you can look at this ?

**Comment by aslafy-z** [comment]
@STRML would be great if you could merge this !

**Comment by ranbena** [comment]
+1 for this

**Comment by fedevegili** [comment]
+1 for this

**Comment by fedevegili** [comment]
+1 for this

**Comment by ErkanSensei** [comment]
Let's get this in please! This is an important feature

**Comment by aslafy-z** [comment]
This is still needed! #nostale

**Comment by harleylang** [comment]
+1 still needed! Any chance this can be included in a future milestone?

**Comment by STRML** [comment]
Continued in #1248.

**Comment by STRML** [comment]
Published in `1.0.0`!

**Comment by harleylang** [comment]
Thank you thank you thank you!!

**Pull request closed without merging**